### DISCUSSION TITLE
[MagicLink] Is it possible to make shorthands for paths and wiki URLs?

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#2309] [Q&A]
Question is in the title. Is it possible with the MagicLink extension to turn a link pointing to a file/folder in a repo or the repo's wiki into a shorthand?

For example:
- https://github.com/facelessuser/pymdown-extensions/blob/main/docs/src/markdown/extensions/magiclink.md becomes [facelessuser/pymdown-extensions@`docs/src/markdown/extensions/magiclink.md`](https://github.com/facelessuser/pymdown-extensions/blob/main/docs/src/markdown/extensions/magiclink.md) or something similar
- https://github.com/facelessuser/pymdown-extensions/wiki becomes [facelessuser/pymdown-extensions/wiki](https://github.com/facelessuser/pymdown-extensions/wiki) or something similar.

If possible already, would it allow to define custom classes (i.e. `github-wiki`) to target in CSS for styling?

If not possible, could such a thing be considered?

On another note would I suggest disabling the wiki as it is currently accessible AND can be edited by non-contributors, offering a place for spam.

**Comment by facelessuser** [comment]
MagicLink only supports emalis, URLs, and repository issues, diffs, commits, and discussions.

Does GitHub itself have a special syntax for wiki links? If not, I am doubtful we will be creating our own. And when I say special syntax, I mean from within normal Markdown, not special wiki syntax within wiki sites.

Our current conventions are based on what either GitHub, GitLab, or BitBucket support. The only one we got a little creative with was repository names which we piggybacked off of the user mention format and extended to allow repositories if they came right after the user with a slash: `@facelessuser/pymdown-extensions`.

**Reply by Andre601** [reply]
There isn't any special formatting for wiki links.

**Reply by facelessuser** [reply]
I think, at this time, I'm not particularly interested in creating a syntax for wiki links. If at some future time, GitHub creates a wiki shorthand, I would be happy to consider it, but I don't think I want to roll my own approach that could change if GitHub decides to roll their own.

**Reply by Andre601** [reply]
Could it at least be considered to implement a way to define your own (regex) patterns to use here?

Perhaps it could be added as a `type` option for `custom` that would include some extra settings to define here.
```js
'custom': {
    'example': {
        'type': 'regex',
        'pattern': 'https:\/\/example\.com\/(?<user>[a-zA-Z0-9]+)\/(?<repo>[a-zA-Z0-9_-]+)',
        'result': '$<user>/$<repo>',
        'label': 'Example'
}
```
I am, of course, open for alternative ways here.

**Reply by facelessuser** [reply]
I'm not sure I'm interested in creating and maintaining a fragile way for users to create custom link shorteners. Magilink currently allows shortening and converting those shortened forms to actual links. This would break that convention and would be one way, and I'm not really interested in providing a custom solution to change that. Even the new "custom" format I provide works in the existing framework and just allows for different hosts.
